src/envs/playroom.py

```diff
@@ -0,0 +1,631 @@
+                return True
+        else:  # W
+            if rotation >= 1.5 or rotation <= -1.5:
+                return True
+        return False
+
+    @staticmethod
+    def _On_holds(state: State, objects: Sequence[Object]) -> bool:
+        block1, block2 = objects
+        cls = PlayroomEnv
+        if state.get(block1, "held") >= cls.held_tol or \
+           state.get(block2, "held") >= cls.held_tol:
+            return False
+        x1 = state.get(block1, "pose_x")
+        y1 = state.get(block1, "pose_y")
+        z1 = state.get(block1, "pose_z")
+        x2 = state.get(block2, "pose_x")
+        y2 = state.get(block2, "pose_y")
+        z2 = state.get(block2, "pose_z")
+        return np.allclose([x1, y1, z1], [x2, y2, z2+CFG.playroom_block_size],
+                           atol=cls.pick_tol)
+
+    @staticmethod
+    def _OnTable_holds(state: State, objects: Sequence[Object]) -> bool:
+        block, = objects
+        z = state.get(block, "pose_z")
+        cls = PlayroomEnv
+        desired_z = cls.table_height + CFG.playroom_block_size * 0.5
+        return (state.get(block, "held") < cls.held_tol) and \
+            (desired_z-cls.pick_tol < z < desired_z+cls.pick_tol)
+
+    # To pull in when I add all the predicates
+    # @staticmethod
+    # def _NextToTable_holds(state: State, objects: Sequence[Object]
+    #                        ) -> bool:
+    #     # for now this also includes all table coords
+    #     robot, = objects
+    #     x, y = state.get(robot, "pose_x"), state.get(robot, "pose_y")
+    #     cls = PlayroomEnv
+    #     return (cls.table_x_lb-cls.table_tol < x
+    #             < cls.table_x_ub+cls.table_tol) and \
+    #            (cls.table_y_lb-cls.table_tol < y
+    #             < cls.table_y_ub+cls.table_tol)
+
+    @staticmethod
+    def _NextToDoor_holds(state: State, objects: Sequence[Object]
+                          ) -> bool:
+        robot, door = objects
+        x, y = state.get(robot, "pose_x"), state.get(robot, "pose_y")
+        door_x, door_y = state.get(door, "pose_x"), state.get(door, "pose_y")
+        cls = PlayroomEnv
+        return (door_x-cls.door_tol < x < door_x+cls.door_tol) \
+                and (door_y-cls.door_r-cls.door_tol < y
+                     < door_y+cls.door_r+cls.door_tol)
+
+    # To pull in when I add all the predicates
+    # @staticmethod
+    # def _NextToDial_holds(state: State, objects: Sequence[Object]
+    #                       ) -> bool:
+    #     robot, dial = objects
+    #     x, y = state.get(robot, "pose_x"), state.get(robot, "pose_y")
+    #     dial_x, dial_y = state.get(dial, "pose_x"), state.get(dial, "pose_y")
+    #     cls = PlayroomEnv
+    #     return (dial_x-cls.dial_r-cls.dial_tol < x
+    #             < dial_x+cls.dial_r+cls.dial_tol) and \
+    #            (dial_y-cls.dial_r-cls.dial_tol < y
+    #             < dial_y+cls.dial_r+cls.dial_tol)
+
+    def _Pick_policy(self, state: State, objects: Sequence[Object],
```

Comment:
I'm mostly ignoring this stuff since it'll be for the next PR when you add predicates and options, but I'm also wondering if we can inherit more here